contracts/storage/ServiceAgreementStorage.sol

```diff
@@ -0,0 +1,503 @@
+            submissionsIdx++;
+            epochCommits[submissionsIdx] = commit;
+
+            nextIdentityId = commit.nextIdentity;
+        }
+
+        return epochCommits;
+    }
+
+    function submitCommit(
+        address assetContract,
+        uint256 tokenId,
+        bytes memory keyword,
+        uint8 hashingFunctionId,
+        uint16 epoch,
+        uint96 prevIdentityId
+    )
+        public
+        returns (uint256)
+    {
+        bytes32 agreementId = _generateAgreementId(assetContract, tokenId, keyword, hashingFunctionId);
+
+        require(isCommitWindowOpen(agreementId, epoch), "Commit window is closed!");
+
+        IdentityStorage identityStorage = IdentityStorage(hub.getContractAddress("IdentityStorage"));
+        uint96 identityId = identityStorage.identityIds(keccak256(abi.encodePacked(msg.sender)));
+
+        ProfileStorage profileStorage = ProfileStorage(hub.getContractAddress("ProfileStorage"));
+
+        ScoringProxy scoringProxy = ScoringProxy(hub.getContractAddress("ScoringProxy"));
+        uint32 score = scoringProxy.callScoringFunction(
+            serviceAgreements[agreementId].scoringFunctionId,
+            hashingFunctionId,
+            profileStorage.getNodeId(identityId),
+            keyword,
+            profileStorage.getStake(identityId)
+        );
+
+        _insertCommitAfter(
+            agreementId,
+            epoch,
+            prevIdentityId,
+            CommitSubmission({
+                identityId: identityId,
+                nextIdentity: 0,
+                score: score
+            })
+        );
+
+        // emit CommitSubmitted(
+        //     assetContract,
+        //     tokenId,
+        //     keyword,
+        //     hashingFunctionId,
+        //     identityId,
+        //     profileStorage.getNodeId(identityId),
+        //     score
+        // );
+
+        ParametersStorage parametersStorage = ParametersStorage(hub.getContractAddress("ParametersStorage"));
+        uint256 epochLength = parametersStorage.epochLength();
+        // Returns start time of the proof phase
+        return (
+            serviceAgreements[agreementId].startTime +
+            epochLength * (epoch - 1) +
+            epochLength * serviceAgreements[agreementId].proofWindowOffsetPerc / 100
+        );
+    }
+
+    function isProofWindowOpen(bytes32 agreementId, uint16 epoch)
+        public
+        view
+        returns (bool)
+    {
+        uint256 timeNow = block.timestamp;
+        ServiceAgreement storage agreement = serviceAgreements[agreementId];
+
+        ParametersStorage parametersStorage = ParametersStorage(hub.getContractAddress("ParametersStorage"));
+
+        uint256 proofWindowOffset = agreement.epochLength * agreement.epochsNum * agreement.proofWindowOffsetPerc / 100;
+        uint256 proofWindowDuration = agreement.epochLength * agreement.epochsNum * parametersStorage.proofWindowDurationPerc() / 100;
```

Comment:
perhaps we don't need to initialize these variables if we can just use them below